app/auth/dev-auth.js

```diff
@@ -7,26 +7,39 @@ const {
 } = require("./permissions");
 const { developerName, developerUsername } = require("../config");
 
-const devAccount = {
-  name: developerName || "Developer",
-  username: developerUsername || "developer@defra.gov.uk",
+const getDevAccount = (userId) => {
+  if (userId) {
+    return {
+      name: `Developer-${userId}`,
+      username: `developer+${userId}@defra.gov.uk`,
+    };
+  }
+
+  return {
+    name: developerName || "Developer",
+    username: developerUsername || "developer@defra.gov.uk",
+  };
 };
 
-const getAuthenticationUrl = () => {
+const getAuthenticationUrl = (userId) => {
+  if (userId) {
+    return `/dev-auth?userId=${userId}`;
+  }
+
   return "/dev-auth";
 };
 
-const authenticate = async (_, cookieAuth) => {
+const authenticate = async (userId, cookieAuth) => {
   cookieAuth.set({
     scope: [administrator, processor, user, recommender, authoriser],
-    account: devAccount,
+    account: getDevAccount(userId),
   });
 };
 
 const refresh = async (_account, cookieAuth) => {
   cookieAuth.set({
     scope: [administrator, processor, user, recommender, authoriser],
-    account: devAccount,
+    account: getDevAccount(),
```

Comment:
Is this OK to call through with no param? If this is ever called it will change authentication underneath you from the specific dev account you formerly chose to the default one, which would be a bit odd. Maybe save the last one used in a variable somewhere to use in this case.

If it's never called (maybe the dev one never times out...?) then maybe it's better jus to make this function a no-op instead.

Reply:
yeah good point - I knew it wasnt ideal but figured it would just log you in as the base Developer. I'll add a variable like you suggest.